grails-app/domain/org/pih/warehouse/product/Product.groovy

```diff
@@ -453,6 +454,43 @@ class Product implements Comparable, Serializable {
         }
     }
 
+    InventoryItem getDefaultInventoryItem() {
+        return getInventoryItem(Constants.DEFAULT_LOT_NUMBER)
+    }
+
+
+    InventoryItem getInventoryItem(String lotNumber) {
+        return getInventoryItem(lotNumber, null)
+    }
+
+    // FIXME we could also just traverse the inventoryItems association rather than making an extra query
+    // FIXME but i wanted to move towards a detached criteria / named query fetch
+    // FIXME the expiration date should be involved here, but we don't require it in other places
+    //  so i'll leave that for as a problem for future me
+    InventoryItem getInventoryItem(String lotNumber, Date expirationDate) {
+
+        // Treat the default lot number (DEFAULT) as null
+        if (Constants.DEFAULT_LOT_NUMBER.equalsIgnoreCase(lotNumber)) {
+            lotNumber = null
+        }
+
+        // Find an inventory that matches the provided lot number
+        return InventoryItem.createCriteria().get {
+            and {
+                eq("product", this)
+                if (lotNumber) {
+                    eq("lotNumber", lotNumber)
+                } else {
+                    or {
+                        isNull("lotNumber")
+                        eq("lotNumber", "")
+                    }
+                }
+            }
+        } as InventoryItem
+    }
```

Comment:
This is an issue I mentioned on the tech-huddle channel but I guess it will be best if I left a comment here under the PR to have some code reference

There are some cases that this chunck on code will result in an Exception.
`createCriteria().get` will throw an exception when it encounters more than one result found by given criteria, which is what I have encountered.

running the following query will return a list of products that will cause this issue.

```SQL
select ii.product_id, count(ii.id) from inventory_item ii 
where ii.lot_number is NULL or ii.lot_number = ''
group by ii.product_id
HAVING count(ii.id) > 1
ORDER BY count(*) desc;
```

basically after inspecting these inventory items I have noticed that there are some items that have `lotNumber: NULL` and some that `lotNumber: ""`

**InventoryItem** has a following GORM validation set on lotNumber
```
lotNumber(nullable: true, unique: ['product'], maxSize: 255)
```
so there should be unique lotNumber per product.
As for DB constraints I don't see any fro `inventory_item` table  `lot_number` column